environment/src/main/java/com/sequenceiq/environment/credential/service/CredentialService.java

```diff
@@ -179,12 +179,12 @@ public Credential create(Credential credential, @Nonnull String accountId, @Nonn
         credential.setResourceCrn(credentialCrn);
         credential.setCreator(creatorUserCrn);
         credential.setAccountId(accountId);
+        Credential verifiedCredential = credentialAdapter.verify(credential, accountId).getCredential();
+        if (verifiedCredential.getVerificationStatusText() != null) {
+            throw new BadRequestException(verifiedCredential.getVerificationStatusText());
+        }
         try {
             return transactionService.required(() -> {
```

Comment:
couldn't we just drop the whole transaction?
I'm not sure about the UMS call, but that could take up some time as it's an external call. If we can skip it it would be great.

but I would definitely move the notification out from the transaction